### ISSUE TITLE
[Gen  AI] GPT-2 model vanilla experiments. 

### ISSUE THREAD

**Issue by head-iie-vnr** [Vignana-Jyothi/kp-learnings#11]
- [ ] Download GPT-2
- [ ] Experiment text generation

**Comment by head-iie-vnr** [comment]
With simple code like, we could generate the output. 

```
from transformers import GPT2LMHeadModel, GPT2Tokenizer

model_name = 'gpt2'  # or 'gpt2-medium', 'gpt2-large', 'gpt2-xl'
model = GPT2LMHeadModel.from_pretrained(model_name)
tokenizer = GPT2Tokenizer.from_pretrained(model_name)


def generate_text(prompt, max_length=50):
    inputs = tokenizer.encode(prompt, return_tensors='pt')
    outputs = model.generate(inputs, max_length=max_length, num_return_sequences=1)
    text = tokenizer.decode(outputs[0], skip_special_tokens=True)
    return text

prompt = "Once upon a time"
generated_text = generate_text(prompt)
print(generated_text)
```


output

> Once upon a time, the world was a place of great beauty and great danger. The world was a place of great danger, and the world was a place of great danger. The world was a place of great danger, and the world was a

**Comment by head-iie-vnr** [comment]
**Warnings**
> The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results. Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
> The attention mask is not set and cannot be inferred from input because pad token is same as eos token.As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.

**Explanation**
Setting pad_token_id: This is done to ensure that padding tokens are properly handled during generation.
Providing attention_mask: This ensures that the model can correctly focus on the input tokens without getting confused by padding tokens.

Fixed through code fixes.

**Comment by head-iie-vnr** [comment]
By adding an interactive method we can convert it into chat interface.  But the GPT Model is not well trained yet. 

```
def chatbot():
    while True:
        user_input = input("You: ")
        if user_input.lower() in ["exit", "quit"]:
            break
        response = generate_text(user_input)
        print("Bot:", response)

```
python3 chat_interaction.py 

> You: Hello
> Bot: Hello, I'm sorry, but I'm not sure if you're aware of this. I'm not sure if you're aware of this. I'm not sure if you're aware of this. I'm not sure if you're aware of this
> You: I love india
> Bot: I love india, but I'm not sure if I can live in it. I'm not sure if I can live in it. I'm not sure if I can live in it. I'm not sure if I can live in it.
> You: I would like to go to Himalayas
> Bot: I would like to go to Himalayas and visit the Himalayas. I would like to go to the Himalayas and visit the Himalayas. I would like to go to the Himalayas and visit the Himalayas.
> You: Fruits are healthy
> Bot: Fruits are healthy, and they're not just for the kids. They're also for the adults.
> 
> "We're not going to be able to do that," he said. "We're going to have to do something about it."
>
